packages/x-date-pickers/src/tests/describeGregorianAdapter/testCalculations.ts

```diff
@@ -164,122 +277,343 @@ export const testCalculations: DescribeGregorianAdapterTestSuite = ({ adapter })
     });
   });
 
-  it('Method: isEqual', () => {
-    expect(adapter.isEqual(adapter.date(null), null)).to.equal(true);
-    expect(adapter.isEqual(testDateIso, adapter.date(TEST_DATE_ISO_STRING))).to.equal(true);
-    expect(adapter.isEqual(null, testDateIso)).to.equal(false);
-    expect(adapter.isEqual(testDateLocale, adapter.date(TEST_DATE_LOCALE_STRING))).to.equal(true);
-    expect(adapter.isEqual(null, testDateLocale)).to.equal(false);
+  describe('Method: isEqual', () => {
+    it('should work in the same timezone', () => {
+      expect(adapter.isEqual(adapter.date(null), null)).to.equal(true);
+      expect(adapter.isEqual(testDateIso, adapter.date(TEST_DATE_ISO_STRING))).to.equal(true);
+      expect(adapter.isEqual(null, testDateIso)).to.equal(false);
+      expect(adapter.isEqual(testDateLocale, adapter.date(TEST_DATE_LOCALE_STRING))).to.equal(true);
+      expect(adapter.isEqual(null, testDateLocale)).to.equal(false);
+    });
+
+    it('should work with different timezones', function test() {
+      if (!adapter.isTimezoneCompatible) {
+        this.skip();
+      }
+
+      const londonTimezone = adapterTZ.setTimezone(testDateIso, 'Europe/London');
```

Comment:
Nitpick: This reads a bit ambiguously. 🙈 
By reading this line, I'd expect the `londonTimezone` to be just that, some sort of meta information about this particular timezone. 🤷 
What do you think about changing the method name to `setDateTimezone`? 🤔 
Also, update the `londonTimezone` to `dateWithLondonTZ` or `dateWithLondonTimezone`.

Reply:
Hm... I double-checked the existing adapter methods and it looks like all of them omit the `Date` part in the method name. 🤔 
Maybe sticking to the same convention is the best approach..? 🤔 
I would maybe rephrase my comment to clarify that the variable name confused me the most. 🙈 
What do you think about only renaming the variables to better express what kind of value they hold?

Reply:
`dateInLondonTZ` would work for you ?

Reply:
Sounds great 👌 🙈